src/clients/rust/tests/tests.rs

```diff
@@ -0,0 +1,302 @@
+use anyhow::Result as AnyResult;
+use futures::executor::block_on;
+use std::process::{Child, Command};
+use std::sync::atomic::{AtomicU16, Ordering};
+use std::{env, fs};
+use tempfile::TempDir;
+
+use tigerbeetle as tb;
+
+const FIRST_PORT_NUMBER: u16 = 20_000;
+static NEXT_PORT_NUMBER: AtomicU16 = AtomicU16::new(FIRST_PORT_NUMBER);
+
+struct TestHarness {
+    tigerbeetle_bin: String,
+    temp_dir: TempDir,
+    port_number: u16,
+    server: Option<Child>,
+}
+
+impl TestHarness {
+    fn new(name: &str) -> AnyResult<TestHarness> {
+        let manifest_dir = env!("CARGO_MANIFEST_DIR");
+
+        let tigerbeetle_bin = if cfg!(unix) {
+            format!("{manifest_dir}/../../../tigerbeetle")
+        } else if cfg!(windows) {
+            format!("{manifest_dir}/../../../tigerbeetle.exe")
+        } else {
+            todo!()
+        };
+
+        let work_dir = format!("{manifest_dir}/work");
+        fs::create_dir_all(&work_dir)?;
+        let temp_dir = TempDir::with_prefix_in(name, &work_dir)?;
+
+        let port_number = NEXT_PORT_NUMBER.fetch_add(1, Ordering::SeqCst);
+        assert!(port_number != 0);
+
+        Ok(TestHarness {
+            tigerbeetle_bin,
+            temp_dir,
+            server: None,
+            port_number,
+        })
+    }
+
+    fn prepare_database(&self) -> AnyResult<()> {
+        let mut cmd = Command::new(&self.tigerbeetle_bin);
+        cmd.current_dir(self.temp_dir.path());
+        cmd.args([
+            "format",
+            "--replica-count=1",
+            "--replica=0",
+            "--cluster=0",
+            "0_0.tigerbeetle",
+        ]);
+        let status = cmd.status()?;
+
+        assert!(status.success());
+
+        Ok(())
+    }
+
+    fn serve(&mut self) -> AnyResult<()> {
+        assert!(self.server.is_none());
+
+        let mut cmd = Command::new(&self.tigerbeetle_bin);
+        cmd.current_dir(self.temp_dir.path());
+        cmd.args([
+            "start",
+            &format!("--addresses=127.0.0.1:{}", self.port_number),
+            "0_0.tigerbeetle",
+        ]);
+        let child = cmd.spawn()?;
+        self.server = Some(child);
+
+        Ok(())
+    }
+}
+
+impl Drop for TestHarness {
+    fn drop(&mut self) {
+        if let Some(mut child) = self.server.take() {
+            child.kill().expect("kill");
+            let _ = child.wait().expect("wait");
+        }
+    }
+}
+
+#[test]
+fn smoke() -> AnyResult<()> {
+    let ledger = 1;
+    let code = 1;
+
+    let account_id1 = 1;
+    let account_id2 = 2;
+    let transfer_id1 = 3;
+
+    let account_id2_user_data_32 = 4;
+    let transfer_id1_user_data_32 = 5;
+
+    block_on(async {
+        let mut harness = TestHarness::new("smoke")?;
+        harness.prepare_database()?;
+        harness.serve()?;
+
+        let address = &format!("127.0.0.1:{}", harness.port_number);
+        let client = tb::Client::new(0, address)?;
+
+        {
+            let res = client
```

Comment:
`let result`, we use fully-named variables! 

(FWIW, I used to be in the `res` camp before TigerBeetle :)

Reply:
Done.